transport/src/main/java/io/netty/channel/CoalescingBufferQueue.java

```diff
@@ -37,11 +37,16 @@
 public final class CoalescingBufferQueue {
 
     private final Channel channel;
-    private final ArrayDeque<Object> bufAndListenerPairs = new ArrayDeque<Object>();
+    private final ArrayDeque<Object> bufAndListenerPairs;
     private int readableBytes;
 
     public CoalescingBufferQueue(Channel channel) {
+        this(channel, 16);
```

Comment:
This was just to preserve the previous default (`CoalescingBufferQueue` could be used in places other than codec-http2). I would be OK with changing this to something else (it was never explicitly made part of the `CoalescingBufferQueue` interface contract) but now that there is a constructor which explicitly takes `initialSize` the user can set it to w/e they want (and doesn't have to rely on a default). WDYT?